### PR TITLE
Introducing include and exclude in zip action

### PR DESCRIPTION
**Pull request by pritidesai**
Closes #991 

- Adding support for `include` and `exclude` in zip action:

```
packages:
    zipactionwithinclude:
        actions:
            greeting1:
                function: actions/greeting1
                runtime: nodejs:6
                include:
                    - ["actions/common/utils.js", "common/utils.js"]
            greeting2:
                function: actions/greeting2
                runtime: nodejs:6
                include:
                    - ["actions/common/utils.js", "common/utils.js"]
```

For the directory structure:

```
ls -1 actions/common/
utility.js
ls -1 actions/greeting1/
index.js
package.json
ls -1 actions/greeting2/
index.js
package.json
```

Exclude Use Case:

```
packages:
    zipactionwithexclude:
        actions:
            greeting1:
                function: actions
                runtime: nodejs:6
                exclude:
                    - actions/*
                include:
                    - ["actions/common/utils.js", "common/utils.js"]
                    - ["actions/index.js", "index.js"]
                    - ["actions/package.json", "package.json"]
```

For the directory structure:

```
ls -1 actions/
common/
greeting/
index.js
libs/
package.json

ls -1 actions/common/
utils.js

ls -1 actions/greeting/
index.js
package.json

ls -1 actions/libs/
lib1/
lib2/
lib3/
```

Added two integration test `zipactionwithinclude` and `zipactionwithexclude`.

### CODE REVIEW

File: parsers/manifest_parser.go [hunk 1]
```diff
@@ -649,7 +649,7 @@ func (dm *YAMLParser) readActionFunction(manifestFilePath string, manifestFileNa
 
 	if utils.IsDirectory(actionFilePath) {
 		zipFileName = actionFilePath + "." + runtimes.ZIP_FILE_EXTENSION
-		err := utils.NewZipWritter(actionFilePath, zipFileName).Zip()
```
Comment (mrutkows) [inline parsers/manifest_parser.go]: is there a reason this is called "Writter" and not 'Writer" ?

File: parsers/manifest_parser.go [hunk 2]
```diff
@@ -649,7 +649,7 @@ func (dm *YAMLParser) readActionFunction(manifestFilePath string, manifestFileNa
...
+		err := utils.NewZipWritter(actionFilePath, zipFileName, action.Include, action.Exclude, filepath.Dir(manifestFilePath)).Zip()
```
Comment (mrutkows) [inline parsers/manifest_parser.go]: good, this keeps a pipeline concept going for the future...

File: utils/file.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
+/*
```
Comment (mrutkows) [inline utils/file.go]: Thanks for splitting out this module.

**Comment by musa-pro** [comment]
I was following your article at medium - 
https://medium.com/openwhisk/whisk-deploy-zip-actions-with-include-exclude-30ba6d96ad8b
But I couldn't make it working. I followed exactly what you did except my manifest.yml file is in root, not under actions. Could you put all your codes in a repo so I can try that to deploy in ibm cloud if that works or not.

**Comment by mrutkows** [comment]
@musa-pro @pritidesai Hi Mohammad, If you believe there is a bug in the utility, please open an issue and include your manifest contents and any other config. info (.wskconfig file for example).  It is not customary to comment on merged PRs and expect any action to be taken.  In addition, I am not sure how old the blog is, but there are testcases within the repo. that test this functionality with every change to the code.  You can likely find one or more within the https://github.com/apache/openwhisk-wskdeploy/tree/master/tests/src/integration directory structure.

**Pull request merged**